test/e2e/nbthread_test.go

```diff
@@ -0,0 +1,128 @@
+// +build e2e
+
+package e2e
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"testing"
+	"time"
+
+	operatorv1 "github.com/openshift/api/operator/v1"
+
+	"github.com/openshift/cluster-ingress-operator/pkg/operator/controller"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+)
+
+const nbthreadRouterDeployTimeout time.Duration = 1 * time.Minute
+
+// NOTE: This test will mutate the default ingresscontroller and the
+// "cluster" ingress configuration.
```

Comment:
Is this e2e test actually modifying the cluster ingress config?

Reply:
It is not; it only modifies the default ingresscontroller. The comment has been updated